app/src/lib/git/rebase.ts

```diff
@@ -91,48 +92,46 @@ export async function getRebaseContext(
  * and it will probably have a different commit history.
  *
  * @param baseBranch the ref to start the rebase from
- * @param featureBranch the ref to rebase onto `baseBranch`
+ * @param targetBranch the ref to rebase onto `baseBranch`
  */
 export async function rebase(
   repository: Repository,
   baseBranch: string,
-  featureBranch: string
-) {
-  return await git(
-    ['rebase', baseBranch, featureBranch],
+  targetBranch: string
+): Promise<RebaseResult> {
+  const result = await git(
+    ['rebase', baseBranch, targetBranch],
     repository.path,
     'rebase',
     { expectedErrors: new Set([GitError.RebaseConflicts]) }
   )
+
+  return parseRebaseResult(result)
 }
 
 /** Abandon the current rebase operation */
 export async function abortRebase(repository: Repository) {
   await git(['rebase', '--abort'], repository.path, 'abortRebase')
 }
 
-export enum ContinueRebaseResult {
+export enum RebaseResult {
   CompletedWithoutError = 'CompletedWithoutError',
   ConflictsEncountered = 'ConflictsEncountered',
   OutstandingFilesNotStaged = 'OutstandingFilesNotStaged',
   Aborted = 'Aborted',
 }
 
-const rebaseEncounteredConflictsRe = /Resolve all conflicts manually, mark them as resolved/
-
-const filesNotMergedRe = /You must edit all merge conflicts and then\nmark them as resolved/
-
-function parseRebaseResult(result: IGitResult): ContinueRebaseResult {
+function parseRebaseResult(result: IGitResult): RebaseResult {
   if (result.exitCode === 0) {
-    return ContinueRebaseResult.CompletedWithoutError
+    return RebaseResult.CompletedWithoutError
   }
 
-  if (rebaseEncounteredConflictsRe.test(result.stdout)) {
```

Comment:
I bumped `dugite` to the latest version to replace this regex inspection with proper error codes for the known rebase problem scenarios: https://github.com/desktop/dugite/pull/306